### DISCUSSION TITLE
Got InvalidCastException: Failed casting content to 'MyCustomPart', check you have registered your content part with AddContentPart while trying to querying orchard core content items

### DISCUSSION THREAD

**Discussion by PaariMobimp** [OrchardCMS/OrchardCore#17004] [Q&A]
Looking from the tutorial video [here](https://www.youtube.com/watch?v=IYKEeYxeNck&t=1004s), I am trying to fetch my content items. I already have a content type which includes a custom content part as shown below:
![contentType](https://github.com/user-attachments/assets/1e3b749d-6bf5-4d13-85cc-21ad633a9e9e)

Below image is my custom part
![contentPart](https://github.com/user-attachments/assets/80a1331d-db1d-4dc1-9f51-dee55003bb0c)

And the controller code is below:
![controllerCode](https://github.com/user-attachments/assets/4959999e-2837-47da-b332-c05dc371f5ca)

MyCustomPart :
![mycustomPartCode](https://github.com/user-attachments/assets/74cbeb42-368f-4753-91d6-0547fab04f21)

startup.cs code:
![startup.cs](https://github.com/user-attachments/assets/cd4d6d70-39f4-4092-b287-550eb4c14715)

Can someone  help how do I overcome this exception even after adding addContentPart under Services() in my startup file

**Comment by Piedone** [comment]
This is a `Startup` class in your module, right? Is the module turned on (from the Feature admin screen) too?

You don't need `AddRazorPages` and `AddControllers` there BTW.
